examples/src/main/java/org/apache/ignite/examples/datagrid/store/CacheLoadOnlyStoreExample.java

```diff
@@ -0,0 +1,192 @@
+            }
+            finally {
+                // Distributed cache could be removed from cluster only by #destroyCache() call.
+                ignite.destroyCache(CACHE_NAME);
+            }
+        }
+    }
+
+    /**
+     * Creates cache configurations for the loader.
+     *
+     * @return {@link CacheConfiguration}.
+     */
+    private static CacheConfiguration cacheConfiguration(ProductLoader productLoader) {
+        CacheConfiguration cacheCfg = new CacheConfiguration();
+
+        cacheCfg.setCacheMode(CacheMode.PARTITIONED);
+        cacheCfg.setName(CACHE_NAME);
+
+        cacheCfg.setReadThrough(true);
+        cacheCfg.setWriteThrough(true);
+        cacheCfg.setLoadPreviousValue(true);
+
+        // provide the loader.
+        cacheCfg.setCacheStoreFactory(new FactoryBuilder.SingletonFactory(productLoader));
+
+        return cacheCfg;
+    }
+
+    /**
+     * Csv data loader for product data.
+     */
+    private static class ProductLoader extends CacheLoadOnlyStoreAdapter<String, Product, String> implements Serializable {
+        /** Csv file name. */
+        final String csvFileName;
+
+        /** Constructor. */
+        ProductLoader(String csvFileName) {
+            this.csvFileName = csvFileName;
+        }
+
+        /** {@inheritDoc} */
+        @Override protected Iterator<String> inputIterator(@Nullable Object... args) throws CacheLoaderException {
+            Scanner scanner;
+
+            try {
+                scanner = new Scanner(new File(this.getClass().getClassLoader().getResource(csvFileName).getFile()));
+
+                scanner.useDelimiter("\\n");
+            }
+            catch (FileNotFoundException e) {
+                throw new CacheLoaderException("Failed to open the source file " + csvFileName, e);
+            }
+
+            /**
+             * Iterator for text input. The scanner is implicitly closed when there's nothing to scan.
+             */
+            return new Iterator<String>() {
+                /** {@inheritDoc} */
+                @Override public boolean hasNext() {
+                    if (!scanner.hasNext()) {
+                        scanner.close();
+
+                        return false;
+                    }
+
+                    return true;
+                }
+
+                /** {@inheritDoc} */
+                @Override public String next() {
+                    if (!hasNext())
+                        throw new NoSuchElementException();
+
+                    return scanner.next();
+                }
+            };
+        }
+
+        /** {@inheritDoc} */
+        @Nullable @Override protected IgniteBiTuple<String, Product> parse(String rec, @Nullable Object... args) {
+            String[] p = rec.split("\\s*,\\s*");
+            return new T2<>(p[0], new Product(p[0], p[1], p[2], p[3], new BigDecimal(p[4])));
+        }
+    }
+
+    /**
+     * Sample product class.
+     */
+    private static class Product {
```

Comment:
By consistency reasons I would use an already existed model class in this example. A lot of examples use org.apache.ignite.examples.model.Person. However you can use any other model object from this package.

Are you ok with this?